src/particles/distribution/SpinvMF.H

```diff
@@ -0,0 +1,146 @@
+/* Copyright 2022-2023 The Regents of the University of California, through Lawrence
+ *           Berkeley National Laboratory (subject to receipt of any required
+ *           approvals from the U.S. Dept. of Energy). All rights reserved.
+ *
+ * This file is part of ImpactX.
+ *
+ * Authors: Chad Mitchell, Axel Huebl
+ * License: BSD-3-Clause-LBNL
+ */
+#ifndef IMPACTX_DISTRIBUTION_SPINVMF
+#define IMPACTX_DISTRIBUTION_SPINVMF
+
+#include <ablastr/constant.H>
+
+#include <AMReX_Math.H>
+#include <AMReX_Random.H>
+#include <AMReX_REAL.H>
+
+#include <cmath>
+
+
+namespace impactx::distribution
+{
+    struct SpinvMF
+    {
+        /** A von Mises-Fisher (vMF) distribution on the unit 2-sphere.
+         *  This is used for initializing particle spin.  There is a natural
+         *  bijective correspondence between vMF distributions and mean
+         *  (polarization) vectors.
+         *
+         * The algorithm used here is a simplification of the algorithm described in:
+         * C. Pinzon and K. Jung, "Fast Python sampler of the von Mises Fisher distribution", in the special
+         * case of the 2-sphere.  Additional references used include:
+         *
+         * K. V. Mardia and P. E. Jupp, Directional Statistics, Wiley, 1999;
+         * S. Kang and H-S. Oh, "Novel sampling method for the von Mises-Fisher distribution", Stat.
+         * and Comput. 34, 106 (2024), https://doi.org/10.1007/s11222-024-10419-3.
+         *
+         * Return sampling from a vMF distribution.
+         *
+         * @param mux,muy,muz components of a unit vector specifying the mean direction
+         * @param kappa concentration parameter
+         */
+        SpinvMF (
+            amrex::ParticleReal mux,
+            amrex::ParticleReal muy,
+            amrex::ParticleReal muz,
+            amrex::ParticleReal kappa
+        )
+        : m_muX(mux), m_muY(muy), m_muZ(muz), m_kappa(kappa)
+        {
+        }
```

Comment:
Since we can calculate `kappa` via
```C++
kappa = distribution::SpinvMF::inverse_Langevin(pmag);
```
Do we really need to ask the "SpinvMF user" about it? Or can we do:
```suggestion
        SpinvMF (
            amrex::ParticleReal mux,
            amrex::ParticleReal muy,
            amrex::ParticleReal muz
        )
        : m_muX(mux), m_muY(muy), m_muZ(muz)
        {
            amrex::ParticleReal const pmag = std::sqrt(polarization_x*polarization_x+polarization_y*polarization_y+polarization_z*polarization_z);
            m_kappa = distribution::SpinvMF::inverse_Langevin(pmag);
        }
```

Reply:
Will push this change as well.